### PR TITLE
Node split

### PR DESCRIPTION
**Pull request by Delfil**
Adds filtering on nodes.
Adds combining of nodes.
Reduces the memory usage of nodes.
Moves GraphParser to GraphData.

### CODE REVIEW

File: src/main/java/tudelft/ti2806/pl3/data/graph/GraphData.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+public class GraphData {
+	private List<Node> nodes;
+	private List<Edge> edges;
+	private List<Genome> genomes;
+	
+	public GraphData(List<Node> nodes, List<Edge> edges, List<Genome> genomes) {
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/data/graph/GraphData.java]: Missing some javadoc

File: src/main/java/tudelft/ti2806/pl3/visualization/DisplayView.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.graphstream.graph.Graph;
+import org.graphstream.graph.implementations.SingleGraph;
+
+import tudelft.ti2806.pl3.data.graph.Edge;
+import tudelft.ti2806.pl3.data.graph.Node;
+
+import java.util.List;
+
+public class DisplayView {
+	public static Graph getGraph(String name, List<Node> nodes, List<Edge> edges) {
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/visualization/DisplayView.java]: Missing javadoc

File: src/main/java/tudelft/ti2806/pl3/visualization/GraphModel.java [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+	 * @return a list of all edges of nodes with only one output
+	 */
+	protected List<Edge> findToEdges(List<Edge> edges) {
+		sortEdgesOnTo(edges);
+		List<Edge> foundEdges = new ArrayList<Edge>();
+		Edge lastEdge = null;
+		boolean found = true;
+		for (Edge edge : edges) {
+			if (lastEdge != null
+					&& edge.getTo().getNodeId() == lastEdge.getTo().getNodeId()) {
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/visualization/GraphModel.java]: Line longer than 100 characters

**Comment by TomBrouws** [comment]
Looks good, except for Mathieu's comments. Can be merged

**Pull request merged**